### PR TITLE
updating code

### PR DESCRIPTION
**Pull request by KwassiSenam**
I need to update Python to 3.10 to use 'match' statements.

### CODE REVIEW

File: src/server/app.py [hunk 1]
```diff
@@ -0,0 +1,78 @@
+from flask import Flask
+import logging
+#from flask_restful import Api, Resource, reqparse
+from flask_cors import CORS
+#from FlaskApp.api_layer.HelloApiHandler import HelloApiHandler
+from flask import abort, render_template, request, redirect, url_for, jsonify, json
```
Comment (Axelvel) [inline src/server/app.py:6]: Add these functions to fix the server-side cross-site scripting vulnerability
```suggestion
from flask import abort, render_template, request, redirect, url_for, jsonify, json, make_response, escape
```

File: src/server/app.py [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+import numpy as np
+
+app = Flask(__name__)
+#app = Flask(__name__, static_url_path='', static_folder='react_client/build')
+CORS(app) #comment this on deployment
+#api = Api(app)
+
+@app.route('/test', methods = ['POST'] )
+def hello():
+    return request.form['hyper_params']
```
Comment (Axelvel) [inline src/server/app.py:19]: Change this line to fix the server-side cross-site scripting vulnerability #9 
```suggestion
    return make_response(escape(request.form['hyper_params']))
```

**Comment by galyfray** [comment]
A more descriptive PR would be appreciated. What is the match statement, and why do you need it ?

File: src/server/DRL_parameters.py [hunk 3]
```diff
@@ -0,0 +1,25 @@
+class DRL_parameters:
+    def __init__(self,
+        nb_epochs: int,
+        learning_rate: float,
+        batch_size: int,
+        momentum: float) :
+        self.__nbEpochs = nb_epochs
```
Comment (galyfray) [inline src/server/DRL_parameters.py:7]: In python, the norm is to name attributes using the sanke_case. For example `__nbEpocks` gives `__nb_epochs`

File: src/server/DRL_parameters.py [hunk 4]
```diff
@@ -0,0 +1,25 @@
+class DRL_parameters:
```
Comment (galyfray) [inline src/server/DRL_parameters.py:1]: Please add a class doc string.

File: src/server/DRL_parameters.py [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+        nb_epochs: int,
+        learning_rate: float,
+        batch_size: int,
+        momentum: float) :
+        self.__nbEpochs = nb_epochs
+        self.__learningRate = learning_rate
+        self.__batchSize = batch_size
+        self.__momentum = momentum
+
+    def get_parameters(self):
```
Comment (galyfray) [inline src/server/DRL_parameters.py:12]: Why not replacing this method with the built-in magic method `__dict__` ? Is there a specific implementation purpose for this ?

File: src/server/DRL_parameters.py [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+        return {
+            "nbEpochs" : self.__nbEpochs,
+            "learningRate" : self.__learningRate,
+            "batchSize" : self.__batchSize,
+            "momentum" : self.__momentum
+        }
+    
+    def set_parameters(self, params):
```
Comment (galyfray) [inline src/server/DRL_parameters.py:20]: It would be better to split this method in multiple smaller ones.

File: src/server/GA_parameters.py [hunk 7]
```diff
@@ -0,0 +1,25 @@
+class GA_parameters:
```
Comment (galyfray) [inline src/server/GA_parameters.py:1]: Please add a class doc string. Python's class names should be written using UpperCamelCase

File: src/server/GA_parameters.py [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+        nb_epochs: int,
+        pop_size: int,
+        crossover_rate: float,
+        mutation_rate: float) :
+        self.__nbEpochs = nb_epochs
+        self.__popSize = pop_size
+        self.__crossoverRate = crossover_rate
+        self.__mutationRate = mutation_rate
+
+    def get_parameters(self):
```
Comment (galyfray) [inline src/server/GA_parameters.py:12]: This method is duplicated with the DRL parameter class. What is the purpose of using classes here ? If using a class is motivated, consider using polymorphism

File: src/server/Utils/Datasets/E-n29-k4-s7-c200-ecap100.csv [hunk 9]
```diff
@@ -0,0 +1,113 @@
+﻿ID;NODE_COORD_X;NODE_COORD_Y;DEMAND
```
Comment (galyfray) [inline src/server/Utils/Datasets/E-n29-k4-s7-c200-ecap100.csv:1]: Some information are missing using this storage form. Where do you store it ?

File: src/server/Utils/utils.py [hunk 10]
```diff
@@ -0,0 +1,67 @@
+import numpy as np
+
+def choose_data(nb):
```
Comment (galyfray) [inline src/server/Utils/utils.py:3]: You may want to use an array for this. If you store all data sets names in an array, then selecting the nth datasets can be simply done by `ds_array[n]`. This is functionally the same, but will help with testing and will make the code more readable.

File: src/server/Utils/utils.py [hunk 11]
```diff
@@ -0,0 +1,67 @@
...
+        case 22:
+            dataset_choice = "X-n830-k171-s11-c358-ecap1385.csv"
+        case 23:
+            dataset_choice = "X-n920-k207-s4-c33-ecap2773.csv"
+        case _:
+            dataset_choice = "E-n29-k4-s7-c200-ecap100.csv"
+
+    return dataset_choice
+
+def read_data(filename):
```
Comment (galyfray) [inline src/server/Utils/utils.py:59]: Type hints are missing.

**Review by galyfray — CHANGES_REQUESTED** [review]
Please conform to the agreed coding style. Note that the `__init__.py` file is useful. Take a look at the Codacy's bot returns as most of them are revelant.

File: src/server/app.py [hunk 12]
```diff
@@ -0,0 +1,78 @@
...
+        'x': 1, 'y': 10}, {'x': 2, 'y': 13}, {'x': 3, 'y': 18}, {'x': 4, 'y': 20},
+        {'x': 5, 'y': 17},{'x': 6, 'y': 10}, {'x': 7, 'y': 13}, 
+        {'x': 8, 'y': 18}, {'x': 9, 'y': 20}, {'x': 10, 'y': 17}]; 
+    return dps;
+
+# main driver function
+if __name__ == '__main__':
+    # run() method of Flask class runs the application
+    # on the local development server.
+    app.run(debug=True)
```
[the remark this answers is not in the document]
Comment (KwassiSenam) [reply src/server/app.py:78]: this parameter is used in development mode and will be removed for production use.

File: src/server/app.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (KwassiSenam) [reply src/server/app.py:19]: The security aspect will be taken into account in the final step of development.

File: src/server/app.py (hunk 1 above)
Comment (KwassiSenam) [reply src/server/app.py:6]: Ok thanks Axel.

File: src/server/DRL_parameters.py (hunk 3 above)
Comment (KwassiSenam) [reply src/server/DRL_parameters.py:7]: You're right. I just wanted to code the parameters and attributes differently, so I will reverse it.

File: src/server/DRL_parameters.py (hunk 4 above)
Comment (KwassiSenam) [reply src/server/DRL_parameters.py:1]: It is not the final version. It's just to let you see what I will send to the solver.

File: src/server/DRL_parameters.py (hunk 6 above)
Comment (KwassiSenam) [reply src/server/DRL_parameters.py:20]: why ? 
I don't need it in my part-code personally. But it can be an improvement point, no problem.

**Comment by KwassiSenam** [comment]
@galyfray As I say it is not the final version. It is just a plan code to let you see what I will send to the solver. I will take all your comments and suggestions into account in the future step.

File: src/server/DRL_parameters.py (hunk 6 above)
Comment (galyfray) [reply src/server/DRL_parameters.py:20]: > why ?

For several reason :
- If someone ever need to change a single parameter at a time he will need to change all of them
- If you add another parameter, all calls to this methods will have to be updated
- For a better separation of concerns : a single method handling everything is not always the best approach. The limits between what should and what should be separated is an open discussion :)

**Comment by galyfray** [comment]
> I will take all your comments and suggestions into account in the future step.

Thank you

>As I say it is not the final version. It is just a plan code to let you see what I will send to the solver. 

Well, code in the master branch should be quite close to its final form. Functions should be documented, implemented methods should be as close to done as possible, code should be tested and conform to the code style. Creating a PR to push to master is raising a flag saying your code is ready to be shipped. The master branch is not the place for "plan code". This is why I took your code as a final product and review it as is. Many things need to be changed and as of now I think that your plan isn't much readable through the lines of code. Or at least I can't. I don't know what the others think, but another way may be better, to show your plan before or alongside implementing it. :)

**Comment by KwassiSenam** [comment]
Oh sure it is my fault. I didn't need to do a pr, it is a mistake. As I say, just consider the output or what i send to the solver and continue your coding with.

File: src/server/DRL_parameters.py (hunk 6 above)
Comment (KwassiSenam) [reply src/server/DRL_parameters.py:20]: ok we will discuss it tomorrow.

File: src/server/Utils/Datasets/E-n29-k4-s7-c200-ecap100.csv (hunk 9 above)
Comment (KwassiSenam) [reply src/server/Utils/Datasets/E-n29-k4-s7-c200-ecap100.csv:1]: Good point. I will add the missing information asap.

**Comment by galyfray** [comment]
It has been stated in person that this PR can be safely closed.

**Pull request closed without merging**